app/classifier/task.jsx

```diff
@@ -0,0 +1,299 @@
+  <p className="back-button-warning" >Going back will clear your work for the current task.</p>
+
+class Task extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleAnnotationChange = this.handleAnnotationChange.bind(this);
+    this.addAnnotationForTask = this.addAnnotationForTask.bind(this);
+    this.completeClassification = this.completeClassification.bind(this);
+    this.destroyCurrentAnnotation = this.destroyCurrentAnnotation.bind(this);
+    this.warningToggleOn = this.warningToggleOn.bind(this);
+    this.warningToggleOff = this.warningToggleOff.bind(this);
+    this.state = {
+      BackButtonWarning: false
+    }
+  }
+  
+  componentWillMount() {
+    const {workflow, classification} = this.props;
+    classification.annotations = classification.annotations ? classification.annotations : [];
+    if (classification.annotations.length === 0) {
+      this.addAnnotationForTask(workflow.first_task);
+    }
+  }
+
+  handleAnnotationChange(newAnnotation) {
+    const {workflow, classification} = this.props;
+    classification.annotations[classification.annotations.length - 1] = newAnnotation;
+    classification.update('annotations');
+  }
+
+  // Next (or first question)
+  addAnnotationForTask(taskKey) {
+    const {workflow, classification} = this.props;
+    const taskDescription = workflow.tasks[taskKey];
+    let annotation = tasks[taskDescription.type].getDefaultAnnotation(taskDescription, workflow, tasks);
+    annotation.task = taskKey;
+
+    if (workflow.configuration.persist_annotations) {
+      const cachedAnnotation = CacheClassification.isAnnotationCached(taskKey);
+      if (cachedAnnotation) {
+        annotation = cachedAnnotation;
+      }
+    }
+
+    classification.annotations.push(annotation);
+    classification.update('annotations');
+  }
+
+  // Done
+  completeClassification() {
+    const {workflow, classification} = this.props;
+    if (workflow.configuration.persist_annotations) {
+      CacheClassification.delete();
+    }
+
+    const currentAnnotation = classification.annotations[classification.annotations.length - 1];
+    const currentTask = workflow.tasks[currentAnnotation.task];
+    !!currentTask && !!currentTask.tools && currentTask.tools.map((tool) => {
+      if (tool.type === 'grid') {
+        GridTool.mapCells(classification.annotations);
+      }
+    });
+    classification.update({
+      completed: true,
+      'metadata.session': getSessionID(),
+      'metadata.finished_at': (new Date()).toISOString(),
+      'metadata.viewport': {
+        width: innerWidth,
+        height: innerHeight
+      }
+    });
+
+    if (currentAnnotation.shortcut) {
+      addAnnotationForTask(currentTask.unlinkedTask);
+      const newAnnotation = classification.annotations[classification.annotations.length - 1];
+      newAnnotation.value = currentAnnotation.shortcut.index;
+      delete currentAnnotation.shortcut;
+    }
+    this. props.completeClassification();
```

Comment:
Looks like an accidental space here, though it's not breaking saving the classification with a test project I have. I still see the classification save being logged in the console. It definitely is breaking in the dev classifier though.